pandas/core/indexes/multi.py

```diff
@@ -243,10 +243,19 @@ def __new__(cls, levels=None, codes=None, sortorder=None, names=None,
 
         if verify_integrity:
             result._verify_integrity()
+
```

Comment:
so this is going to be pretty non-performant this way. rather I would rather have ``verify_integrity`` should just do a check for this.

Reply:
So for NaN level input we should just raise a value error?

Reply:
Thanks for the comment, I have updated the reassignment of na and vectorised it to make it run faster